litgpt/__main__.py

```diff
@@ -0,0 +1,137 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+
+from typing import TYPE_CHECKING, Any
+
+import torch
+
+if TYPE_CHECKING:
+    from jsonargparse import ArgumentParser
+
+
+def _new_parser(**kwargs: Any) -> "ArgumentParser":
+    from jsonargparse import ArgumentParser, ActionConfigFile
+
+    kwargs.setdefault("default_env", True)
+    parser = ArgumentParser(**kwargs)
+    parser.add_argument(
+        "-c", "--config", action=ActionConfigFile, help="Path to a configuration file in json or yaml format."
+    )
+    return parser
+
+
+def main() -> None:
+    from litgpt.pretrain import setup as pretrain_fn
```

Comment:
I didn't test if this is the case, but if `__main__.py` gets automatically imported even if the CLI is not used, I wanted to avoid importing all the modules unnecessarily